src/main/java/com/bobocode/petros/container/ApplicationAnnotationContainer.java

```diff
@@ -17,12 +18,30 @@ public ApplicationAnnotationContainer(String packageName) {
 
     @Override
     public <T> T getDependency(String name, Class<T> clazz) {
-
-        return null;
+        return (T) dependencyMap.values().stream()
+                .filter(obj -> obj.getClass().getName().toLowerCase().equals(name.toLowerCase())
+                        && obj.getClass().equals(clazz))
+                .findFirst();
     }
 
     @Override
     public <T> T getDependency(Class<T> clazz) {
-        return null;
+        if (!isNonUniqueDependency(clazz)){
```

Comment:
Try to avoid negate in _if_ statements. You may throw an exception first or change method to _isUniqueDependency(...)_